<?xml version="1.0"?>
<launch>

	<arg name="Name_Robot" default="RosAria" />
	<arg name="Name_Laser" default="/scan" />
	<arg name="RosAria_port" default="/dev/pioneer/usb_to_serial_port"/>
	<arg name="hokuyo_port" default="/dev/ttyACM0" />
	<arg name="map_file" default="$(find ekf_localization)/ros/map/map_with_edits.yaml"/>
	
	<!--node name="RosAria" pkg="rosaria" node="RosAria">
		<param name="RosAria/port" type="string" value="$(arg RosAria_port)"/>
	</node>

	<node pkg="urg_node" type="urg_node" name="hokuyo_node">
      <param name="ip_address" value=""/>
      <param name="serial_port" value="$(arg hokuyo_port)"/>
      <param name="serial_baud" value="115200"/>
      <param name="frame_id" value="laser"/>
      <param name="calibrate_time" value="true"/>
      <param name="publish_intensity" value="false"/>
      <param name="publish_multiecho" value="false"/>
      <param name="angle_min" value="-1.5707963"/>
      <param name="angle_max" value="1.5707963"/>
    </node-->

	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>
	<node name="laser_tf_broadcaster" pkg="ekf_localization" type="laser_tf_broadcaster.py"/>

 	<!-- small description about your node  args="$(find ekf_localization_name)/ros/config/my_arg_file.yaml" -->
 
 	<node pkg="ekf_localization" type="ekf_localization_node.py" name="ekf_node"
 	respawn="false" output="screen">
		<param name="robot_name" type="string" value="$(arg Name_Robot)"/>
		<param name="laser_topic" type="string" value="$(arg Name_Laser)"/>
		<param name="distance_threshold" value="0.2"/>
		<param name="angle_threshold" value="0.34906585"/>
		<!--param name="belief" value="[0.0, 0.0, 0.0]"/>
		<param name="sigma" value="[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]"/-->
	</node>

</launch>
